executor/runtime/docker/docker.go

```diff
@@ -2959,6 +2959,8 @@ func (r *DockerRuntime) Kill(ctx context.Context, wasKilled bool) error { // nol
 		containerStopTimeout = time.Second * time.Duration(*killWait)
 	}
 	cStopPtr := &containerStopTimeout
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, containerStopTimeout)
```

Comment:
I don't think we want to descend from that context because it. can be cancelled. I think we would want to do `context.WithTimeout(context.Background(), ...)`. This creates a second-order problem that what if it fails?

Reply:
Yea...
I'm not super sure. Seems like we gotta fail *eventually*, but yea, we will leave disconnected containers in their wake?